docs/docs/migrate-from-v1/README.md

```diff
@@ -0,0 +1,259 @@
+:::
+
+### Exiting a workflow early 
+
+In v1, the `$exit` function can be called to exit a flow early:
+
+```javascript
+async (event, steps) {
+  exit('Exiting the whole workflow early');
+  console.log('I will never run');
+}
+```
+
+In v2, this same function is available, but under `$.exit`:
+
+```javascript
+defineComponent({
+  async run({ steps, $ }) {
+    $.flow.exit('Exiting the workflow early');
+    console.log('I will never run');
+  }
+});
+```
+
+::: warning
+In v2, `$end` does not display the reason in the workflow results logs.
+::: 
+
+### Params vs Props
+
+In workflow builder v2, both Component **props** and Node.js code steps **props** are one in the same.
+
+You can still enter free text and select data from other steps in pre-built actions. Also can add your own custom props that accept input like strings, numbers and more just like in v1.
+
+#### Defining params
+
+In the v1 workflow builder, params could be structured or unstructured. The [params schema builder](https://pipedream.com/docs/v1/workflows/steps/params/#configuring-custom-params) allowed you to add your own custom params to steps.
+
+In v2, you can add your own custom props without leaving the code editor with `props`.
+
+```javascript
+export default defineComponent({
+  props: {
+    firstName: {
+      type: 'string',
+      label: 'Your first name',
+    }
+  },
+  async run({ steps, $ }) {
+    console.log(this.firstName);
+  }
+});
+
+```
+
+In the example above a `firstName` string prop is created. The value assigned to this prop in the workflow builder by either a static string or dynamic data will be available in the step as `this.firstName`.
+
+Additionally, a visual component is rendered in the step **Configuration** to accept this input:
+
+<img src="./images/custom-string-prop.png" alt="Custom props render in the Configuration portion of the code step.">
+
+### Connecting apps
+
+In the v2 builder, you can connect apps with your code using [props](/components/props).
+
+Above the `run` function define an app prop that your Node.js step integrates with.
+
+```javascript
+export default defineComponent({
+  props: {
+    twitter: {
+      type: "app",
+      app: "slack",
+    }
+  },
+  async run({ steps, $ }) {
+    // now your Slack token for authenticating API calls will be available here:
+    this.slack.$auth.oauth_access_token;
+  },
+})
+```
+
+After testing the step the Slack app will appear in the **Configuration** section on the left hand side. In this section you can choose which Slack account you'd like to use in the step.
+
+<img src="./images/app-props-example.png" alt="Example of adding an app connection to a v2 Node.js step">
+
+
+### HTTPs Response
+
+You can still return an HTTP response from an HTTP triggered workflow.
+
+Use `$.respond` to send a JSON or string response from the HTTP call that triggered the workflow.
```

Comment:
Can you link to the `$.respond` docs here? Technically you can return other types of responses (e.g. a Buffer), which we include in those docs.